### PR TITLE
Add Gazebo worlds topic

### PR DESCRIPTION
**Pull request by hamishwillee**
This adds a gazebo worlds topic, (mirroring the [gazebo-classic version here](https://docs.px4.io/main/en/sim_gazebo_classic/worlds.html))

The idea is to show the worlds, make it clear what they are intended for etc.

This is missing:
- Images - do we have any available? If not, what is the best way to generate an image? A viewer? Screenshot of running vehicle?
- Statement of what all/each are used for, whether they are used for things by default.

FYI @chfriedrich98 @Claudio-Chies @Jaeyoung-Lim - can you help/advise?

### CODE REVIEW

File: en/sim_gazebo_gz/worlds.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Empty world (a grey plane).
+This is used by default.
+
+[PX4-gazebo-models/main/worlds/default.sdf](https://github.com/PX4/PX4-gazebo-models/blob/main/worlds/default.sdf)
+
+![screenshot of default world](../../assets/simulation/gazebo/worlds/default.png)
+
+## Aruco
+
+TBD: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:21]: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?

File: en/sim_gazebo_gz/worlds.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+[PX4-gazebo-models/main/worlds/aruco.sdf](https://github.com/PX4/PX4-gazebo-models/blob/main/worlds/aruco.sdf)
+
+![screenshot of Aruco world](../../assets/simulation/gazebo/worlds/aruco.png)
+
+## Baylands
+
+Baylands world surrounded by water.
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:29]: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?

File: en/sim_gazebo_gz/worlds.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+[PX4-gazebo-models/main/worlds/bayland.sdf](https://github.com/PX4/PX4-gazebo-models/blob/main/worlds/baylands.sdf)
+
+![screenshot of Baylands world](../../assets/simulation/gazebo/worlds/baylands.png)
+
+## Lawn
+
+TBD: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:37]: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?

How does this differ from Rover world? Which should you use, and when?

File: en/sim_gazebo_gz/worlds.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+[PX4-gazebo-models/main/worlds/lawn.sdf](https://github.com/PX4/PX4-gazebo-models/blob/main/worlds/lawn.sdf)
+
+![screenshot of lawn world](../../assets/simulation/gazebo/worlds/lawn.png)
+
+## Rover
+
+TBD: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:45]: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?

How does this differ from Lawn world? Which should you use, and when?

File: en/sim_gazebo_gz/worlds.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+[PX4-gazebo-models/main/worlds/rover.sdf](https://github.com/PX4/PX4-gazebo-models/blob/main/worlds/rover.sdf)
+
+![screenshot of rover world](../../assets/simulation/gazebo/worlds/rover.png)
+
+## Walls
+
+World for testing collision prevention.
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:53]: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?

File: en/sim_gazebo_gz/worlds.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+[PX4-gazebo-models/main/worlds/walls.sdf](https://github.com/PX4/PX4-gazebo-models/blob/main/worlds/walls.sdf)
+
+![screenshot of walls world](../../assets/simulation/gazebo/worlds/walls.png)
+
+## Windy
+
+[Empty world](#default) with wind enabled.
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:61]: What's it for? What vehicles should it be used with. Is it used by any vehicles by default?

Can we say anything about the wind here - i.e how windy? Anything configurable? Is he wind variable or consistent or noisy. Anything developers might find relevant?

File: en/sim_gazebo_gz/worlds.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+The model specific worlds are:
+
+<!--
+- [boat.world](https://github.com/PX4/PX4-SITL_gazebo-classic/blob/main/worlds/boat.world): Includes a surface to simulate buoyancy of the [boat](../sim_gazebo_classic/vehicles.md#unmanned-surface-vehicle-usv-boat).
+- [uuv_hippocampus.world](https://github.com/PX4/PX4-SITL_gazebo-classic/blob/main/worlds/uuv_hippocampus.world): An empty world used to simulate an underwater environment for the [HippoCampus UUV](../sim_gazebo_classic/vehicles.md#hippocampus-tuhh-uuv).
+- [typhoon_h480.world](https://github.com/PX4/PX4-SITL_gazebo-classic/blob/main/worlds/typhoon_h480.world): Used by [Typhoon H480 (Hexrotor)](../sim_gazebo_classic/vehicles.md#typhoon-h480-hexrotor) vehicle model and includes a video widget to enable / disable video streaming.
+  The world includes a gazebo plugin for a simulated camera.
+- [iris_irlock.world](https://github.com/PX4/PX4-SITL_gazebo-classic/blob/main/worlds/iris_irlock.world): Includes a IR beacon for testing [precision landing](../advanced_features/precland.md).
+-->
```
Comment (hamishwillee) [inline en/sim_gazebo_gz/worlds.md:80]: This section copied from Gazebo-Classic:
- Is this true/same logic for Gazebo?
- If so, what are the items that should be listed? What do they show?

File: en/sim_gazebo_gz/worlds.md (hunk 4 above)
Comment (chfriedrich98) [reply en/sim_gazebo_gz/worlds.md:45]: The rover world is very similiar to the lawn world with these 2 important differences differences:
- Grid on the ground which is useful as a reference while driving.
- Higher update rate which solves segfault issues specifically with rovers with ackermann steering.

Currently the r1 rover (4009) and ackermann rover (4012) use it as default and all future ground vehicles I'll work with will have it as default aswell.
I will most likely add more rover friendly changes to it down the line since there are still some issues with rover simulation.

In summary: It is a world that is and will be further optimized for ground vehicles.

File: en/sim_gazebo_gz/worlds.md (hunk 1 above)
Comment (Claudio-Chies) [reply en/sim_gazebo_gz/worlds.md:21]: Apparently it's for precision landing with a downward facing mono camera. 
https://github.com/PX4/PX4-gazebo-models/pull/48
mabye @dakejahl could help elaborate on this

Comment (dakejahl) [reply en/sim_gazebo_gz/worlds.md:21]: yes it's the same as the default world with the addition of an aruco tag. This is used in conjunction with the x500_mono_cam_down airframe to do precision landing

File: en/sim_gazebo_gz/worlds.md (hunk 4 above)
Comment (hamishwillee) [reply en/sim_gazebo_gz/worlds.md:45]: Thanks. FYI I see a grid on lawn. I do see segfaults, but it worked for ackermann, not r1.

File: en/sim_gazebo_gz/worlds.md (hunk 1 above)
Comment (hamishwillee) [reply en/sim_gazebo_gz/worlds.md:21]: Thanks!

File: en/sim_gazebo_gz/worlds.md (hunk 6 above)
Comment (hamishwillee) [reply en/sim_gazebo_gz/worlds.md:72]: @frede791 Can you advise ^^^ when you might use this/what you'd like to show? Any configuration? I'm going to merge this PR because this shouldn't block, but I know people will want to know if the world is suitable before they try it.

**Comment by hamishwillee** [comment]
Thanks all. I'm merging as this is a good starting point. Hopefully we can expand a little on things like windy world in future.
